### DISCUSSION TITLE
Hide certain nested Composables at capture

### DISCUSSION THREAD

**Discussion by BenjyTec** [PatilShreyas/Capturable#70] [Q&A]
I am trying to hide certain nested Composables before capturing their parent. I tried it by using a variable like this:
`var isCapturing by remember { mutableStateOf(false) }`

The nested Composable that should be hidden on a Capture looks like this:
`if (!isCapturing) {
    IconButton(
        ....
    ) {
        ....
    }
}`

Then, I trigger the Capture like this:
`Button(
    onClick = { 
        isCapturing = true; 
        captureController.capture() 
    }
) {
        Text(text = "Save image")
}`

However, the element that should be hidden still appears on the resulting bitmap. It seems like before the element can hide, the capture() command is executed, that seems to block the Recomposition.

How could I solve this in an elegant way? I am thankful for any ideas and explanations.

**Comment by faraz152** [comment]
set your flag then wait one frame before capturing so compose has time to hide those views:

```kotlin
scope.launch {
    isCapturing = true
    withFrameNanos { } // lets compose draw the updated frame first
    val bitmap = captureController.captureAsync().await()
    isCapturing = false
}
```

without the `withFrameNanos` the capture fires before the recomposition happens.
